src/cmd/troubleshoot/builder.go

```diff
@@ -88,27 +92,89 @@ func (builder CommandBuilder) buildRun() func(*cobra.Command, []string) error {
 		}
 
 		apiReader := k8scluster.GetAPIReader()
-		tests := []troubleshootFunc{
+		dynakubeUnrelatedTests := []troubleshootFunc{
 			checkNamespace,
-			checkDynakube,
-			checkDTClusterConnection,
-			checkImagePullable,
 		}
+
 		troubleshootCtx := troubleshootContext{
 			context:       context.Background(),
 			apiReader:     apiReader,
 			httpClient:    &http.Client{},
 			namespaceName: namespaceFlagValue,
-			dynakubeName:  dynakubeFlagValue,
 		}
-
-		for _, test := range tests {
+		for _, test := range dynakubeUnrelatedTests {
 			err = test(&troubleshootCtx)
 			if err != nil {
 				logErrorf(err.Error())
 				return nil
 			}
 		}
+		resetLog()
+
+		perDynakubeTests := []troubleshootFunc{
+			checkDynakube,
+			checkDTClusterConnection,
+			checkImagePullable,
+		}
+
+		troubleshootCtx.dynakubeName = dynakubeFlagValue
+		dynakubes, err := getDynakubes(troubleshootCtx)
+		if err != nil {
+			return nil
+		}
+
+		for _, dynakube := range dynakubes {
+			troubleshootCtx = troubleshootContext{
+				context:       context.Background(),
+				apiReader:     apiReader,
+				httpClient:    &http.Client{},
+				namespaceName: namespaceFlagValue,
+				dynakube:      dynakube,
+				dynakubeName:  dynakube.Name,
+			}
+			logNewDynakubef(troubleshootCtx.dynakubeName)
+			for _, test := range perDynakubeTests {
+				err = test(&troubleshootCtx)
+				if err != nil {
+					logErrorf(err.Error())
+					return nil
+				}
+			}
+			resetLog()
+			logOkf("'%s' - all checks passed", troubleshootCtx.dynakubeName)
```

Comment:
```suggestion
			}
			
			logNewDynakubef(troubleshootCtx.dynakubeName)
			
			for _, test := range perDynakubeTests {
				err = test(&troubleshootCtx)
				if err != nil {
					logErrorf(err.Error())
					return nil
				}
			}
			
			resetLog()
			logOkf("'%s' - all checks passed", troubleshootCtx.dynakubeName)
```
use more negative space to group different kinds of statements